src/com/esotericsoftware/kryo/util/Util.java

```diff
@@ -26,6 +26,9 @@
 /** A few utility methods, mostly for private use.
  * @author Nathan Sweet <misc@n4te.com> */
 public class Util {
+	//See: https://stackoverflow.com/questions/3038392/do-java-arrays-have-a-maximum-size
+	static public final int maxSafeArraySize = Integer.MAX_VALUE - 8;

```

Comment:
Oh yeah, you're right, I should have made this final - I don't think that I intended this to be modifiable (I really hope that nobody wants to do this) ;-)